sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -196,6 +208,34 @@ case class RLike(left: Expression, right: Expression)
   override def escape(v: String): String = v
   override def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).find(0)
   override def toString: String = s"$left RLIKE $right"
+
+  override protected def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val patternClass = classOf[Pattern].getName
+    val pattern = ctx.freshName("pattern")
+
+    nullSafeCodeGen(ctx, ev, (eval1, eval2) => {
+      val patternCode =
+        if (right.foldable) {
```

Comment:
I think we should move this out of `nullSafeCodeGen`, or it will also eval the `eval2`.
```
if (right.foldable) {
  val rVal = right.eval()
  if (rVal != null) {
  } else {
   "boolean {ev.isNull} = true;  xxx"
  }
} else {
  nullSafeCodeGen(xxx)
}
```